src/screens/add-experience/add-experience.tsx

```diff
@@ -1,39 +1,68 @@
 import React from 'react';
-import { useNavigation } from '@react-navigation/native';
+import { RouteProp, useNavigation, useRoute } from '@react-navigation/native';
 import { ScrollView, View } from 'react-native';
 import { SafeAreaView } from 'react-native-safe-area-context';
 import { Formik } from 'formik';
 
-import { ButtonMode, HeadingLevel } from 'src/common/enums';
+import { ButtonMode, HeadingLevel, ProfileRoute } from 'src/common/enums';
 import { FormDate, FormInput, Heading, MainButton } from 'src/components';
 import { addExperienceValidationSchema } from 'src/validation-schemas';
 import { experienceActions } from 'src/store/experience';
-import { IAddCareer } from 'src/common/types';
+import { IAddCareer, ProfileParamList } from 'src/common/types';
 import { useAppDispatch } from 'src/hooks';
 import { defaultAddExperiencePayload } from './common';
 import styles from './styles';
 
+type AddExperienceRouteProps = RouteProp<
+  ProfileParamList,
+  ProfileRoute.ADD_CAREER_EXPERIENCE
+>;
+
 const AddExperienceScreen: React.FC = () => {
   const dispatch = useAppDispatch();
   const navigation = useNavigation();
+  const route = useRoute<AddExperienceRouteProps>();
+  const { isEdit, career } = route.params;
 
   const handleCancel = () => {
     navigation.goBack();
   };
 
   const handleAddExperience = (values: IAddCareer) => {
-    dispatch(
-      experienceActions.addCareerExperience({
-        position: values.position,
-        company: values.company,
-        startDate: values.startDate,
-        endDate: values.endDate,
-      })
-    );
+    if (isEdit) {
+      dispatch(
+        experienceActions.editCareerExperience({
+          id: career.id,
+          position: values.position,
+          company: values.company,
+          startDate: values.startDate,
+          endDate: values.endDate,
+        })
+      );
+    } else {
+      dispatch(
+        experienceActions.addCareerExperience({
+          position: values.position,
+          company: values.company,
+          startDate: values.startDate,
+          endDate: values.endDate,
+        })
+      );
+    }
```

Comment:
Maybe it's better to change it to something like this
```suggestion
    const commonPayload = {
          position: values.position,
          company: values.company,
          startDate: values.startDate,
          endDate: values.endDate,
    };
    
    if (isEdit) {
      dispatch(
        experienceActions.editCareerExperience({
          id: career.id,
          ...commonPayload
        })
      );
    } else {
      dispatch(
        experienceActions.addCareerExperience(commonPayload)
      );
    }
```